serverless/aws/protomaps-template.yaml

```diff
@@ -0,0 +1,187 @@
+
+  ProtomapsLambdaFunction:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      FunctionName: protomaps
+      Runtime: nodejs18.x
+      Architectures: [arm64]
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Handler: index.handler
+      MemorySize: 512
+      Environment:
+        Variables:
+          BUCKET: !Ref BucketName
+          PUBLIC_HOSTNAME: !Ref PublicHostname
+      Code:
+        S3Bucket: !Ref CodeBucketName
+        S3Key: !Ref CodeKey
+
+  ProtomapsLambdaFunctionUrl:
+    Type: 'AWS::Lambda::Url'
+    Properties:
+      AuthType: NONE
+      TargetFunctionArn: !GetAtt ProtomapsLambdaFunction.Arn
+      Cors:
+        AllowOrigins: ["*"]
+      InvokeMode: BUFFERED
+
+  ProtomapsLambdaFunctionUrlPermission:
+    Type: 'AWS::Lambda::Permission'
+    Properties:
+      Action: lambda:InvokeFunctionUrl
+      FunctionName: !Ref ProtomapsLambdaFunction
+      Principal: '*'
+      FunctionUrlAuthType: NONE
+
+# ##########################################################################
+# #  CloudFront::Distribution                                              #
+# ##########################################################################
+
+  CloudFrontDistribution:
+    Type: 'AWS::CloudFront::Distribution'
+    Properties:
+      DistributionConfig:
+        Origins:
+          - Id: ProtomapsLambdaOrigin
+            DomainName: !Select [2, !Split ["/", !GetAtt ProtomapsLambdaFunctionUrl.FunctionUrl]]
+            CustomOriginConfig:
+              OriginProtocolPolicy: https-only
+        DefaultCacheBehavior:
+          TargetOriginId: ProtomapsLambdaOrigin
+          ViewerProtocolPolicy: redirect-to-https
+          CachePolicyId: !Ref CachePolicyId
+          ResponseHeadersPolicyId: !Ref ResponseHeadersPolicyId
+        Enabled: true
+        HttpVersion: http2and3
+        Comment: "Protomaps CloudFront Distribution"
+        PriceClass: PriceClass_All  # Change this to save cost and distribute to fewer countries. Check https://aws.amazon.com/cloudfront/pricing/
+
+# ##########################################################################
+# #  CloudFront::CachePolicy                                               #
+# ##########################################################################
+
+  CachePolicyId:
+    Type: 'AWS::CloudFront::CachePolicy'
+    Properties:
+      CachePolicyConfig:
+        Name: 'CachingOptimized'
+        DefaultTTL: 86400
+        MaxTTL: 31536000
+        MinTTL: 0
+        ParametersInCacheKeyAndForwardedToOrigin:
+          EnableAcceptEncodingBrotli: true
+          EnableAcceptEncodingGzip: true
+          HeadersConfig:
+            HeaderBehavior: none
+          CookiesConfig:
+            CookieBehavior: none
+          QueryStringsConfig:
+            QueryStringBehavior: none
+
+  ResponseHeadersPolicyId:
+    Type: 'AWS::CloudFront::ResponseHeadersPolicy'
+    Properties:
+      ResponseHeadersPolicyConfig:
+        Name: 'protomaps-cors'
```

Comment:
Can this be given a scoped name like `!Sub "${AWS::StackName}-ResponseHeadersPolicy"`? otherwise it looks like it will conflict if there is two stacks using this template.